EIPS/eip-3155.md

```diff
@@ -0,0 +1,241 @@
+- Move the specification to a more visible place to encourage new clients to implement it
+- Strictly define corner cases that were not addressed in the previous version
+- Allow for updates to the specification in case new fields are introduced during execution
+- Provide sample output
+
+Implementing this EIP in all major clients allows us to create meaningful differential fuzzers that fuzz EVM implementations for the mainnet and all upcoming hardforks.
+It also helps finding differences in execution quickly in the case of a chain split.
+
+Due to the slight differences in output the following fields currently need to be dropped when comparing tracing outputs: $GasCost$, $MemorySize$, $RefundCounter$, $ReturnStack$, and $ReturnData$. This reduces the usefulness of tracing significantly. Creating a common standard would allow to add these fields thus increasing the surface that can be tested by a fuzzer.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+Clients should be able to execute simple transactions as well as code and return traces. In the following, we will call this client CUT (client under test) and use go-ethereums `evm` binary for code examples. 
+
+### Inputs
+
+The CUT MUST be able to execute a statetest as described [here](https://www.ethdocs.org/en/latest/contracts-and-transactions/ethereum-tests/state_tests/index.html).
+Example :
+``` 
+evm --json --nomemory statetest foobar.json 
+```
+The CUT MUST spit out jsonl as defined below, and it MUST also report the final `stateRoot` hash after executing the test. The CUT SHOULD report whether the tests succeeded or failed.
+
+If the statetest contains a Fork identifier (e.g. `Berlin` or `Istanbul`) the CUT MUST run the statetest under the chain rules of this fork.
+
+The CUT SHOULD executed statetests if the filenames are written to STDIN. This allows to execute multiple tests on the same instance without having to set up and tear down the client.
+
+### Output
+
+The CUT MUST output a `json` object for EACH operation. The output SHOULD to be printed to STDERR.
+
+Required Fields: 
+- `pc` Program Counter
+- `op` OpCode
+- `gas` Gas left before executing this operation
+- `gasCost` Gas cost of this operation
+- `stack` Array of all values on the stack
+- `depth` Depth of the call stack
+- `returnStack` Array of values, Stack of the called function
+- `returnData` Data returned by function call
+- `refund` Amount of **global** gas refunded
+- `memSize` Size of memory array
+
+Optional Fields: 
+- `opName` Name of the operation
+- `error` Description of an error 
+- `memory` Array of all allocated values
+- `storage` Array of all stored values
+
```

Comment:
I think it would be worthwhile to say that clients SHOULD output the fields in the order they are listed in this EIP.